File: go/testdata/fixture-multi-lang/services/jvm-suite/scala/PlainUtils.scala
package com.example.utils

object PlainUtils {
  def add(a: Int, b: Int): Int = a + b

  def greet(name: String): String = s"Hello, $name!"

  def reverseList[A](xs: List[A]): List[A] = xs.reverse
}

class Counter(initial: Int) {
  private var count = initial
  def increment(): Unit = { count += 1 }
  def value: Int = count
}

File: testdata/fixture-multi-lang/services/jvm-suite/kotlin/PlainUtils.kt
package com.example.utils

fun add(a: Int, b: Int): Int = a + b

fun greet(name: String): String = "Hello, $name!"

fun <A> reverseList(xs: List<A>): List<A> = xs.reversed()

class Counter(initial: Int) {
    private var count = initial
    fun increment() { count++ }
    fun value(): Int = count
}

data class Pair<A, B>(val first: A, val second: B)
